src/Estimators/tests/RecursiveLeastSquareTest.cpp

```diff
@@ -0,0 +1,133 @@
+/**
+ * @file RecursiveLeastSquareTest.cpp
+ * @authors Giulio Romualdi
+ * @copyright 2020 Istituto Italiano di Tecnologia (IIT). This software may be modified and
+ * distributed under the terms of the GNU Lesser General Public License v2.1 or any later version.
+ */
+
+#include <cmath>
+#include <random>
+#include <functional>
+
+// Catch2
+#include <catch2/catch.hpp>
+
+// YARP
+#include <yarp/os/Property.h>
+#include <yarp/os/Searchable.h>
+#include <yarp/os/Value.h>
+#include <yarp/os/ResourceFinder.h>
+#include <yarp/os/Bottle.h>
+
+#include <iDynTree/Core/MatrixDynSize.h>
+#include <iDynTree/Core/EigenHelpers.h>
+
+#include <BipedalLocomotionControllers/ParametersHandler/IParametersHandler.h>
+#include <BipedalLocomotionControllers/ParametersHandler/YarpImplementation.h>
+
+#include <BipedalLocomotionControllers/Estimators/RecursiveLeastSquare.h>
+
+#include <ConfigFolderPath.h>
+
+using namespace BipedalLocomotionControllers::Estimators;
+using namespace BipedalLocomotionControllers::ParametersHandler;
+
+class Model
+{
+    iDynTree::Vector2 params;
+    double x;
+
+    std::random_device rd{};
+    std::mt19937 gen;
+
+    std::normal_distribution<> noise1{0,0.5};
+    std::normal_distribution<> noise2{0,0.5};
+
+public:
+    Model(const iDynTree::Vector2& params)
+        : gen{rd()}
+        , params(params)
+    {
+    }
+
+    iDynTree::MatrixDynSize regressor()
```

Comment:
Could you also document this model and the regressor? so when someone reads the test, they know what's the model in question? even if its arbitrarily chosen.

Reply:
Test documentation improved in https://github.com/dic-iit/bipedal-locomotion-controllers/pull/23/commits/a751616f622dbcd2550fdee59500ee4825c80d29